### PR TITLE
feat: add new comparators to bucketing lib

### PR DESCRIPTION
**Pull request by suthar26**
add `startWith` `endWith` and the opposites to the bucketing lib

### CODE REVIEW

File: lib/shared/bucketing/__tests__/segmentation.test.ts [hunk 1]
```diff
@@ -1091,6 +1091,90 @@ describe('SegmentationManager Unit Test', () => {
...
+                values: [''],
+            } as AudienceFilterOrOperator
+            assert.strictEqual(
+                false,
+                segmentation.checkStringsFilter(
+                    'testuser@devcycle.com',
+                    filter,
+                ),
+            )
+        })
```
Comment (JamieSinn) [inline lib/shared/bucketing/__tests__/segmentation.test.ts:1177]: Can we add this for all the new filters and contains?

Comment (suthar26) [reply lib/shared/bucketing/__tests__/segmentation.test.ts:1177]: do you mean add it to `!startWith` `!endWith` ?

Comment (JamieSinn) [reply lib/shared/bucketing/__tests__/segmentation.test.ts:1177]: yep - and the contains operator - just so we have a source of truth with regard to what we expect

Comment (suthar26) [reply lib/shared/bucketing/__tests__/segmentation.test.ts:1177]: added, do you want to make sure its expecting the correct values?

Comment (suthar26) [reply lib/shared/bucketing/__tests__/segmentation.test.ts]: yeah added the comparators

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**